### DISCUSSION TITLE
Are there any differences between Select and Map? if not why include Map in this library

### DISCUSSION THREAD

**Discussion by RollsChris** [louthy/language-ext#922] [Q&A] [answered]
Hi 

I cant see any difference between Select and Map Implementation when reading the docs?

Why include Map? I think it adds more confusion when we already have Select

Thanks

**Comment by StefanBertels** [comment, marked as answer]
This is not my library so maybe the following answer is a guess.

But from what I learned by using the library and sometimes diving into details, the main reason is probably that map, filter, bind, fold etc are common names for those FP operations outside of C#. As a lot of concepts in LanguageExt try to bring FP into "C# world" this makes it a lot easier to understand concepts and to communicate with others.

Probably the name choice in C# (select, where, selectmany, aggregate) was unfortunate from today's FP perspective -- it was heavily inspired by SQL terminology. But that's history. 

For practical reasons LanguageExt types supports C# select, where ...., too. This enables LINQ style -- which is a good fit here, because LINQ _is_ a very functional (FP) concept in C# and it is very useful with LanguageExt.

There might be more technical reasons because of implementation differences between `.Bind` and `.SelectMany` (latter includes projection) or different (more) signatures for `.Map` in comparison to `.Select` (e.g. iterating over (key,value) or (index,value)).

**Comment by RollsChris** [comment]
I kind of thought that would be the case but it does make the landscape a whole lot more confusing lol

**Comment by louthy** [comment]
@StefanBertels is correct, `Select`, `SelectMany`, and `Where` are SQL terminology, not FP terminology.  If I could get away with just `Map`, `Bind`, and `Filter` I would, but `Select`, `SelectMany`, and `Where` are needed to make the monadic types work in LINQ expressions.
